celements-global-redirect/src/main/java/com/celements/globalredirect/GlobalRedirectService.java

```diff
@@ -0,0 +1,101 @@
+package com.celements.globalredirect;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.xwiki.context.Execution;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.execution.XWikiExecutionProp;
+import com.celements.init.XWikiProvider;
+import com.celements.model.reference.RefBuilder;
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiConfigSource;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+@Component
+public class GlobalRedirectService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GlobalRedirectService.class);
+
+  private final AtomicReference<List<GlobalRedirect>> globalRedirectList = new AtomicReference<>();
+
+  private final XWikiProvider xwikiProvider;
+  private final XWikiConfigSource configSource;
+  private final Execution execution;
+
+  @Inject
+  public GlobalRedirectService(XWikiProvider xwikiProvider, XWikiConfigSource configSource,
+      Execution execution) {
+    this.xwikiProvider = xwikiProvider;
+    this.configSource = configSource;
+    this.execution = execution;
+  }
+
+  public boolean isActivated() {
+    return ("1".equals(configSource.getProperty("xwiki.preferences.redirect")));
+  }
+
+  public DocumentReference getGlobalRedirectDocRef() {
+    return RefBuilder.create().wiki("xwiki").doc("XWiki").doc("XWikiPreferences")
+        .build(DocumentReference.class);
```

Comment:
`.doc(...)` twice used.
also use XWikiConstant.MAIN_WIKI, XWIKI_SPACE and XWIKI_PREF_DOC_NAME